crates/core/src/db/datastore/locking_tx_datastore/committed_state.rs

```diff
@@ -373,6 +400,13 @@ impl CommittedState {
 
         self.merge_apply_inserts(&mut tx_data, tx_state.insert_tables, tx_state.blob_store, ctx);
 
+        // If the TX will be logged, record its projected tx offset,
+        // then increment the counter.
+        if self.tx_consumes_offset(&tx_data, ctx) {
+            tx_data.set_tx_offset(self.next_tx_offset);
+            self.next_tx_offset += 1;
```

Comment:
Did you forget to remove the earlier increment?

Reply:
D'oh! Pushed.